napari/_qt/_tests/test_qt_notifications.py

```diff
@@ -192,29 +244,24 @@ def test_notification_error(mock_show, monkeypatch, qtbot):
     except ValueError as e:
         notif = ErrorNotification(e)
 
-    # This test creates TracebackDialog which is not added to qtbot
-    # but its parent is same as parent of NapariQtNotification.
-    # so create dummy parent allow to not leak widget.
-    # Current parent structure, where QWidget is controled by qtbot:
-    # QWidget
-    # ├── NapariQtNotification
-    # └── TracebackDialog
-    widget = QWidget()
-    qtbot.add_widget(widget)
-
-    dialog = NapariQtNotification.from_notification(notif, parent=widget)
+    dialog = NapariQtNotification.from_notification(notif)
 
     bttn = dialog.row2_widget.findChild(QPushButton)
     assert bttn.text() == 'View Traceback'
-    mock_show.assert_not_called()
+    assert count_show.show_traceback_count == 0
     bttn.click()
-    mock_show.assert_called_once()
+    assert count_show.show_traceback_count == 1
 
 
 @pytest.mark.sync_only
-def test_notifications_error_with_threading(make_napari_viewer, clean_current):
+def test_notifications_error_with_threading(
+    make_napari_viewer, clean_current, monkeypatch
+):
     """Test notifications of `threading` threads, using a dask example."""
     random_image = da.random.random((10, 10))
+    monkeypatch.setattr(
+        NapariQtNotification, "show_notification", lambda x: None
+    )
```

Comment:
For me, it is only a workaround, but I do not have any idea how to do this better.

Reply:
👍🏼